addons/calendar/models/calendar.py

```diff
@@ -0,0 +1,1092 @@
+        all_events = self.search([('recurrent_id', '=', self.id), '|', ('active', '=', False), ('active', '=', True)])
+        for ev in all_events:
+            rset1._exdate.append(todate(ev.recurrent_id_date))
+        return [d.astimezone(pytz.UTC) for d in rset1]
+
+    @api.multi
+    def _get_recurrency_end_date(self):
+        data = self.read(['final_date', 'recurrency', 'rrule_type', 'count', 'end_type', 'stop'])[0]
+        if not data.get('recurrency'):
+            return False
+
+        end_type = data.get('end_type')
+        final_date = data.get('final_date')
+        if end_type == 'count' and all(data.get(key) for key in ['count', 'rrule_type', 'stop']):
+            count = data['count'] + 1
+            delay, mult = {
+                'daily': ('days', 1),
+                'weekly': ('days', 7),
+                'monthly': ('months', 1),
+                'yearly': ('years', 1),
+            }[data['rrule_type']]
+
+            deadline = fields.Datetime.from_string(data['stop'])
+            return deadline + relativedelta(**{delay: count * mult})
+        return final_date
+
+    @api.multi
+    def _find_my_attendee(self):
+        """
+            Return the first attendee where the user connected has been invited from all the meeting_ids in parameters
+        """
+        for event in self:
+            for attendee in event.attendee_ids:
+                if self.env.user.partner_id.id == attendee.partner_id.id:
+                    return attendee
+        return False
+
+    @api.multi
+    def get_date_formats(self):
+        lang = self.env.lang
+        lang_params = {}
+        if lang:
+            res_lang = self.env['res.lang'].search([("code", "=", lang)])
+            if res_lang:
+                lang_params = res_lang.read(["date_format", "time_format"])
+
+        # formats will be used for str{f,p}time() which do not support unicode in Python 2, coerce to str
+        format_date = lang_params.get("date_format", '%B-%d-%Y').encode('utf-8')
+        format_time = lang_params.get("time_format", '%I-%M %p').encode('utf-8')
+        return (format_date, format_time)
+
+    @api.multi
+    def get_display_time_tz(self, tz=False):
+        if tz:
+            self.env.context["tz"] = tz
+        event = self[0]
+        return self._get_display_time(event.start, event.stop, event.duration, event.allday)
+
+    @api.model
+    def _get_display_time(self, start, stop, zduration, zallday):
+        """
+            Return date and time (from to from) based on duration with timezone in string :
+            eg.
+            1) if user add duration for 2 hours, return : August-23-2013 at (04-30 To 06-30) (Europe/Brussels)
+            2) if event all day ,return : AllDay, July-31-2013
+        """
+        tz = self.env.context.get('tz')
+        if not tz:  # tz can have a value False, so dont do it in the default value of get !
+            tz = self.env.context.get('tz', self.env.user.tz)
```

Comment:
if not tz then again getting tz from context while tz is already a result from context.get('tz') ?
``` tz = self.env.context('tz') or self.env.user.tz ```